include/swift/AST/Types.h

```diff
@@ -2337,12 +2329,14 @@ class AnyFunctionType : public TypeBase {
     ParameterTypeFlags Flags = {};
     
   public:
-    Type getType() const { return Ty; }
+    Type getType() const;
     CanType getCanType() const {
-      assert(Ty->isCanonical());
-      return CanType(Ty);
+      assert(getType()->isCanonical());
+      return CanType(getType());
     }
     
+    Type getRawType() const { return Ty; }
```

Comment:
`getRawType()` is already used for `Enum` raw types. I wonder if there is a more distinct name that can be used here, e.g. `getPlainType` (I would suggest `getBaseType` but that's already used in different places).